org.bardsoftware.eclipsito/src/org/bardsoftware/impl/eclipsito/BootImpl.java

```diff
@@ -35,6 +39,33 @@ public void run(String application, File modulesDir, String descriptorPattern, L
       return plugins;
     }
 
+  private File getVersionDir(File modulesDir) {
+    File versionFile = new File(modulesDir.getPath() + File.separator + "VERSION");
+    String version = null;
+    try (BufferedReader br = new BufferedReader(new FileReader(versionFile))) {
+      version = br.readLine();
+    } catch (IOException e) {
+      Boot.LOG.severe("No version file found");
+    }
+
+    File additionalVersionFile = new File(System.getProperty("user.home") + File.separator + ".ganttproject.d" + File.separator + "plugins" + File.separator + "VERSION");
+    if (additionalVersionFile.exists()) {
+      try (BufferedReader br = new BufferedReader(new FileReader(versionFile))) {
```

Comment:
Any sense to extract these lines into  a helper function to read the first line?

Reply:
I created this function to split functionality of choosing plugins folder from `run()` method. Because we need to decide which folder to use. And more readable is to do it in a separate function and then return result to `run()`